File: frontend/rentloop_app/lib/features/chat/models/chat_message_dto.dart
class ChatMessageDto {
  final int id;
  final int conversationId;
  final int senderUserId;
  final String senderName;
  final String text;
  final DateTime sentAt;
  final bool isRead;
  final bool isMine;

  ChatMessageDto({
    required this.id,
    required this.conversationId,
    required this.senderUserId,
    required this.senderName,
    required this.text,
    required this.sentAt,
    required this.isRead,
    required this.isMine,
  });

  factory ChatMessageDto.fromJson(Map<String, dynamic> json) {
    return ChatMessageDto(
      id: (json['id'] as num).toInt(),
      conversationId: (json['conversationId'] as num).toInt(),
      senderUserId: (json['senderUserId'] as num).toInt(),
      senderName: (json['senderName'] ?? '').toString(),
      text: (json['text'] ?? '').toString(),
      sentAt: DateTime.parse(json['sentAt'].toString()),
      isRead: json['isRead'] == true,
      isMine: json['isMine'] == true,
    );
  }
}

File: backend/RentLoop/RentLoop.API/DTOs/Chat/ChatMessageDto.cs
namespace RentLoop.API.DTOs.Chat
{
    public class ChatMessageDto
    {
        public int Id { get; set; }
        public int ConversationId { get; set; }

        public int SenderUserId { get; set; }
        public string SenderName { get; set; } = string.Empty;

        public string Text { get; set; } = string.Empty;
        public DateTime SentAt { get; set; }

        public bool IsRead { get; set; }
        public bool IsMine { get; set; } // radi UI-a (mobile/admin)
    }
}
